.buildkite/publish/push-docker.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -exu
+set -o pipefail
+
+function realpath {
+  echo "$(cd "$(dirname "$1")"; pwd)"/"$(basename "$1")";
+}
+
+SCRIPT_DIR=$(realpath "$(dirname "$0")")
+BUILDKITE_DIR=$(realpath "$(dirname "$SCRIPT_DIR")")
+PROJECT_ROOT=$(realpath "$(dirname "$BUILDKITE_DIR")")
+
+if [[ "${ARCHITECTURE:-}" == "" ]]; then
+  echo "!! ARCHITECTURE is not set. Exiting."
+  exit 2
+fi
+
+VERSION_PATH="$PROJECT_ROOT/connectors/VERSION"
+VERSION=$(cat $VERSION_PATH)
+
+if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
+  echo "Adding SNAPSHOT labeling"
+  VERSION="${VERSION}-SNAPSHOT"
+fi
+
+TAG_NAME="docker.elastic.co/enterprise-search/elastic-connectors-${ARCHITECTURE}:${VERSION}"
+
+echo "Loading image from archive file..."
+docker load < "$PROJECT_ROOT/.artifacts/elastic-connectors-docker-${VERSION}-${ARCHITECTURE}.tar.gz"
+
+# ensure +x is set to avoid writing any sensitive information to the console
+set +x
+
+VAULT_ADDR=${VAULT_ADDR:-https://vault-ci-prod.elastic.dev}
+VAULT_USER="docker-swiftypeadmin"
+DOCKER_USER=$(vault read -address "${VAULT_ADDR}" -field user_20230609 secret/ci/elastic-connectors/${VAULT_USER})
+DOCKER_PASSWORD=$(vault read -address "${VAULT_ADDR}" -field secret_20230609 secret/ci/elastic-connectors/${VAULT_USER})
+
+echo "Logging into docker..."
+docker login -u $DOCKER_USER -p $DOCKER_PASSWORD docker.elastic.co
```

Comment:
Same comment on this being insecure and leading to the DOCKER_PASSWORD getting leaked in the buildkite log output. I don't have the syntax handy, but this should also have a way to pipe the password in from STDIN

Reply:
I will look into it (and above as well)... I did at least make sure of the `set +x` and warning to not echo the commands, but if there's a safer way, I'm all for it...